src/game/scripts/vscripts/abilities/pudge_flesh_heap_cooldown_reduction.lua

```diff
@@ -0,0 +1,183 @@
+      end
+
+      local nFXIndex = ParticleManager:CreateParticle( "particles/units/heroes/hero_pudge/pudge_fleshheap_count.vpcf", PATTACH_OVERHEAD_FOLLOW, self:GetCaster() )
+      ParticleManager:SetParticleControl( nFXIndex, 1, Vector( 1, 0, 0 ) )
+      ParticleManager:ReleaseParticleIndex( nFXIndex )
+    end
+  end
+end
+]]
+--------------------------------------------------------------------------------
+
+
+
+--Taken from the spelllibrary, credits go to valve
+
+modifier_flesh_heap_cooldown_reduction = class({})
+
+
+--------------------------------------------------------------------------------
+
+function modifier_flesh_heap_cooldown_reduction:IsHidden()
+    if self:GetAbility():GetLevel() == 0 then
+        return true
+    end
+    return false
+end
+
+--------------------------------------------------------------------------------
+
+function modifier_flesh_heap_cooldown_reduction:RemoveOnDeath()
+    return false
+end
+
+--------------------------------------------------------------------------------
+function modifier_flesh_heap_cooldown_reduction:IsPurgable()
+    return false
+end
+
+--------------------------------------------------------------------------------
+
+function modifier_flesh_heap_cooldown_reduction:IsPassive()
+    return true
+end
+
+function modifier_flesh_heap_cooldown_reduction:IsPurgable()
+    return true
+end
+
+function modifier_flesh_heap_cooldown_reduction:GetFleshHeapKills()
+  if self.nKills == nil then
+    self.nKills = 0
+  end
+  return self.nKills
+end
+ 
+--------------------------------------------------------------------------------
+
+function modifier_flesh_heap_cooldown_reduction:OnCreated( kv )
+  if not self:GetAbility() then
+    self:GetParent():RemoveModifierByName("modifier_flesh_heap_cooldown_reduction")
+    self:GetParent():CalculateStatBonus()
+    return
+  end
+  self.flesh_heap_value_buff_amount = self:GetAbility():GetSpecialValueFor( "flesh_heap_value_buff_amount" ) or 0
+  if IsServer() then
+    self:SetStackCount( self:GetFleshHeapKills() )
+    self:GetParent():CalculateStatBonus()
+  end
+end
+
+--------------------------------------------------------------------------------
+
+function modifier_flesh_heap_cooldown_reduction:OnRefresh( kv )
+  if not self:GetAbility() then
+    self:GetParent():RemoveModifierByName("modifier_flesh_heap_cooldown_reduction")
+    self:GetParent():CalculateStatBonus()
+    return
+  end
+  self.flesh_heap_value_buff_amount = self:GetAbility():GetSpecialValueFor( "flesh_heap_value_buff_amount" ) or 0
+  if IsServer() then
+    self:SetStackCount( self:GetFleshHeapKills() )
+    self:GetParent():CalculateStatBonus()
+  end
+end
+
+--------------------------------------------------------------------------------
+
+function modifier_flesh_heap_cooldown_reduction:DeclareFunctions()
+  local funcs = {
+    MODIFIER_PROPERTY_EVASION_CONSTANT,
```

Comment:
I think "MODIFIER_PROPERTY_EVASION_CONSTANT" should be something like "MODIFIER_PROPERTY_COOLDOWN_PERCENTAGE"